server/controllers/app.controller.js

```diff
@@ -248,17 +248,17 @@ class AppController {
             lastSynchronizationDate: new Date().toISOString(),
           });
 
-        lastStep = 'count-customers';
-        const shopify = this.shopifyClientFactory.createClient(shop, accessToken);
-        const totalCustomers = await shopify.customer.count();
-
         lastStep = 'prepare-customers-list';
+        const shopify = this.shopifyClientFactory.createClient(shop, accessToken);
+        let params = { limit: shopifyCustomersPageSize };
         let customers = [];
-        for (let pageNumber = 1; pageNumber <= totalCustomers/shopifyCustomersPageSize + 1; pageNumber++)
-        {
-          customers = customers.concat((await shopify.customer.list({ limit: shopifyCustomersPageSize, page: pageNumber }))
-            .filter(customer => !!customer.email));
-        }
+        do {
+          console.log('-> Query parameters before api shopify call: ' + params);
```

Comment:
`params` is an object, if you concatenate it to a string, it will be simplified and you will loose information.

```suggestion
          console.log('-> Query parameters before api shopify call', params);
```

Reply:
```suggestion
          // TODO: remove this log, it is to debug before merging
          console.log('-> Query parameters before api shopify call: ' + params);
```